docs/source/commands/subgraphs.mdx

```diff
@@ -255,6 +255,78 @@ If you don't want to wait for the check to complete, you can run the command wit
 
 If you're running schema checks in CI, you might want to pass the `--background` flag to `rover subgraph check`. This flag instructs Rover to initiate schema checks but _not_ await their result. If you've [connected Apollo Studio to your GitHub repository](/graphos/delivery/github-integration/), the integration detects the checks execution and adds a status to the associated pull request.
 
+### `subgraph lint`
+
+<AuthNotice />
+
+You can run the GraphOS schema linter against your local subgraph schema to identify any violations of formatting and naming best practices:
+
+```bash title="Example command"
+rover subgraph lint --name products --schema ./products-schema.graphql my-graph@my-variant
+```
+
+The argument `my-graph@my-variant` in the example above is a **graph ref** that specifies the ID of the graph you're comparing your schema changes against, along with which [variant](/graphos/graphs/#variants) you're comparing against.
+
+> Schema linting also runs as one of the checks included in [`subgraph check`](#subgraph-check). Use this command to perform one-off linting.
+
+Options include:
+
+<table class="field-table">
+<thead>
+<tr>
+<th>Name</th>
+<th>Description</th>
+</tr>
+</thead>
+<tbody>
+<tr class="required">
+<td>
+
+###### `--schema`
+
+</td>
+<td>
+
+**Required.** The path to a local `.graphql` or `.gql` file, in [SDL format](/resources/graphql-glossary/#schema-definition-language-sdl).
+
+Alternatively, you can provide `-`, in which case the command uses an SDL string piped to `stdin` instead (see [Using `stdin`](../conventions#using-stdin)).
+
+</td>
+</tr>
+
+<tr class="required">
+<td>
+
+###### `--name`
+
+</td>
+
+<td>
+
+**Required.**  The name of the published subgraph to compare schema changes against.
```

Comment:
Reading it here, the current behavior could be a little confusing. We would allow providing a subgraph name that hasn't been published to, unless `--ignore-existing-lint-violations` is set.

Reply:
@swcollard So the implication is that the provided `name` only really matters if you set `--ignore-existing-lint-violations`, yeah? But I reckon it's still a required option regardless because it's always required for `subgraph` commands? I see how this option is a bit "vestigial" in most cases.. I'm not sure what clarifying language would be helpful though

Reply:
That's exactly right. I prefer to keep it as a required option for all subgraph commands. But yea it almost feels more confusing than clarifying to explain the edge case.